### PR TITLE
[RFR] Reworked custom Link component to allow react-router links to be used

### PR DESCRIPTION
**Pull request by Tom-Bonnike**
@mattberridge @pgoetze @eschaefer 

Solution so that Links always are a react-router link if they’re not external, and that styles are shared. Tell me what you think.

### CODE REVIEW

File: src/helpers/getPetitionLink.js [hunk 1]
```diff
@@ -1,7 +1,11 @@
...
-    id
-  ].join('/');
+export default (id, fullURL) => {
+  let link = ['petitions', id];
+
+  if (fullURL) {
+    link.unshift(process.env.BASE_URL);
+  } else {
+    link[0] = `/${link[0]}`;
+  }
```
Comment (Tom-Bonnike) [inline src/helpers/getPetitionLink.js]: …

File: src/components/ButtonLink/index.js [hunk 2]
```diff
@@ -1,21 +1,12 @@
...
-  props.external
-    ? <a {...createAttrs(props)}>{props.children || props.text}</a>
-    : <Link {...createAttrs(props)}>{props.children || props.text}</Link>
+  <Link
+    className={getClassname(props)}
+    {...props}
+  />
 );
 
 export default ButtonLink;
```
Comment (Tom-Bonnike) [inline src/components/ButtonLink/index.js]: I guess we could maybe delete that component then…
I just moved the whole `createAttrs` logic to the Link component, see below :)

File: src/helpers/getPetitionLink.js (hunk 1 above)
Comment (mattberridge) [inline src/helpers/getPetitionLink.js]: I would keep these two helpers separate for clarity I think

```
getPetitionPath
getPetitionURL
```

File: src/components/Link/index.js [hunk 3]
```diff
@@ -1,10 +1,23 @@
...
+  props.external
+    ? <a {...createAttrs(props)}>
+      {props.children || props.text || props.href}
+    </a>
+    : <RouterLink {...createAttrs(props)}>
+      {props.children || props.text || props.href}
+    </RouterLink>
 );
 
 export default Link;
```
Comment (mattberridge) [inline src/components/Link/index.js]: This is nice, works well!

File: src/components/ButtonLink/index.js (hunk 2 above)
Comment (mattberridge) [inline src/components/ButtonLink/index.js]: The thing with `<ButtonLink>` is that even though it only uses `<Link>` now it handles the class name construction. Things might be messy if we have to start using `import { getClassname } from 'components/Button'` further up the tree, on `<Link>` directly?

File: src/helpers/getPetitionLink.js (hunk 1 above)
Comment (Tom-Bonnike) [inline src/helpers/getPetitionLink.js]: Oh, Github doesn’t hide the comments on old diffs with the new reviews, but this is done. Yes, good solution. :)

**Pull request merged**